package com.nguyennhatminh614.foodbmianalyzercompose.ui.camera.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.nguyennhatminh614.foodbmianalyzercompose.R
import com.nguyennhatminh614.foodbmianalyzercompose.util.DevicePreview

@Composable
fun CameraNoPermissionScreen(
    onRequestPermissionClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background),
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(
            text = stringResource(R.string.camera_permission_required_desc),
            textAlign = TextAlign.Center,
            color = MaterialTheme.colorScheme.onBackground,
            style = MaterialTheme.typography.bodyLarge,
            modifier = Modifier.padding(16.dp)
        )
        Button(onClick = onRequestPermissionClick) {
            Text(stringResource(R.string.action_grant_permission))
        }
    }
}

@DevicePreview
@Composable
private fun CameraNoPermissionPreview() {
    MaterialTheme {
        Surface {
            CameraNoPermissionScreen(onRequestPermissionClick = {})
        }
    }
}